sonar-ws/src/main/java/org/sonarqube/ws/client/WsClientFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarqube.ws.client;
+
+import org.sonar.api.server.ws.LocalConnector;
+
+/**
+ * @since 5.5
+ */
+public class WsClientFactory {
```

Comment:
An application using the HTTP client has two ways to test:
1. implementing its own own WsClient (which is an interface). Whatever the way to create an instance (Factory with static methods or others), it's up to the application to decouple the usage of WsClient with the way to instantiate it.
2. using a real HTTP server

In the context of SQ plugins defining new web services, `WsTester` should be improved in order to provide the ability to emulate interactions with other WS.